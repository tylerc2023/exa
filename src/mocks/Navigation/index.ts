export const mockNavigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  push: jest.fn(),
  popToTop: jest.fn(),
};
