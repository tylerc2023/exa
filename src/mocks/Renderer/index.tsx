import React from 'react';
import {Provider} from 'react-redux';
import {ReactTestRenderer, create} from 'react-test-renderer';
import {store} from '../../providers/Redux';

export const mockRenderer = (component: JSX.Element): ReactTestRenderer =>
  create(<Provider store={store}>{component}</Provider>);
