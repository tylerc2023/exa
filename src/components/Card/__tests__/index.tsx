import React from 'react';
import {Card} from '..';
import {mockRenderer} from '../../../mocks/Renderer';

it('renders correctly', () => {
  const {tree} = mockRenderer({component: <Card />});
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders correctly with onPress', () => {
  const callback = jest.fn();
  const {tree} = mockRenderer({component: <Card onPress={callback} />});
  expect(tree.toJSON()).toMatchSnapshot();
});
