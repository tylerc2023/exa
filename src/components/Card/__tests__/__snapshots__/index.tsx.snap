// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "hsl(0, 0%, 100%)",
      "borderColor": "hsl(0, 0%, 100%)",
      "borderRadius": 8,
      "borderWidth": 1,
      "elevation": 2,
      "marginVertical": 8,
      "opacity": 1,
      "shadowColor": "hsl(210, 10%, 23%)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.192,
      "shadowRadius": 1.92,
      "zIndex": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "hsla(0,0%,100%,0.07)",
          "borderRadius": 8,
          "padding": 16,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`renders correctly with onPress 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "hsl(0, 0%, 100%)",
      "borderColor": "hsl(0, 0%, 100%)",
      "borderRadius": 8,
      "borderWidth": 1,
      "elevation": 2,
      "marginVertical": 8,
      "opacity": 1,
      "shadowColor": "hsl(210, 10%, 23%)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.192,
      "shadowRadius": 1.92,
      "zIndex": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "hsla(0,0%,100%,0.07)",
          "borderRadius": 8,
          "padding": 16,
        },
        undefined,
      ]
    }
  />
</View>
`;
